Lib/numbers.py

```diff
@@ -290,6 +301,10 @@ def __float__(self):
         """
         return self.numerator / self.denominator
 
+    def is_integer(self):
+        """Does this Rational represent an integral value?"""
+        return self.denominator == 1
```

Comment:
Must `denominator` and `numerator` always be fully reduced?

Reply:
If not,
```suggestion
        return self.numerator % self.denominator == 0
```

Reply:
Yes, `self.numerator` and `self.denominator` will always be relatively prime in normal use (and denominator will always be positive). Other parts of the fractions module assume this, so it's safe to just check `self.denominator == 1` here.

Reply:
This isn't about `Fraction` and its implementation of the Rational API though - this is about whether the API mandates the implementation behaves this way.

Reply:
Ah. I see the doc is `""".numerator and .denominator should be in lowest terms."""` - so perhaps worth clarifying that denominator should always be positive too.

Reply:
Yes, that docstring could definitely be improved. Apart from that clarification, we could do with a line or two describing what the `Rational` class is actually _for_, rather than launching straight into a detail. But that's a job for a separate PR.